src/core/thread/child_table.hpp

```diff
@@ -36,15 +36,16 @@
 
 #include "openthread-core-config.h"
 
-#if OPENTHREAD_FTD
-
 #include "common/iterator_utils.hpp"
 #include "common/locator.hpp"
 #include "common/non_copyable.hpp"
+#include "thread/s2s.hpp"
 #include "thread/topology.hpp"
 
 namespace ot {
 
+#if OPENTHREAD_FTD || OPENTHREAD_MTD_S2S
```

Comment:
Same here, put this line to line 39? We don't want to include these header files if it is not neccessary.